tests/utils_test.py

```diff
@@ -130,11 +139,14 @@ def test_override_param(params):
     param_path_hash = "classes#sex_types#HM#cis_trans"
 
     utils.override_param(params, param_path_hash, "trans", delimiter="#")
-
     assert params.classes.sex_types.HM.cis_trans == "trans"
 
     param_path_pipe = "classes|sex_types|HM|cis_trans"
-
     utils.override_param(params, param_path_pipe, "other")
-
     assert params.classes.sex_types.HM.cis_trans == "other"
+
+    # test if the override works with an int key
```

Comment:
should probably test this still bombs out if an int key is passed but unexpected

Reply:
Wouldn't that be caught with unused params?

Reply:
maybe, but it's a slightly different test - just to make sure the try/except doesn't accidentally swallow that error